service_container/calls.rst

```diff
@@ -71,3 +71,81 @@ To configure the container to call the ``setLogger`` method, use the ``calls`` k
 
         $container->register(MessageGenerator::class)
             ->addMethodCall('setLogger', [new Reference('logger')]);
+
+
+.. versionadded:: 4.3
+
+    The ``immutable-setter`` injection was introduced in Symfony 4.3.
+
+When it comes to optional dependencies, the setter approach can be useful but
+what if you need to change the injected dependency without mutating the initial service?
+
+To do so, you probably could use ``immutable-setter`` injection::
+
+    namespace App\Service;
```

Comment:
Coul we use another namespace without “Service” please? 😁😜

App\Message... ?

Reply:
Improved, I agree, `App\Service` isn't really interesting here 😄